### PR TITLE
use array concatenation instead of push in useFlattenRecords hook

### LINKED ISSUE #983: [bug] useFlattenRecords throws "Maximum call stack size exceeded error"
I have a virtual table and a record with 50k children rows.
"useFlatterRecords" throws "Maximum call stack size exceeded error" while flattening so large number of records.

Same problem described on stackoverflow:
https://stackoverflow.com/questions/61740599/rangeerror-maximum-call-stack-size-exceeded-with-array-push

How to fix:
Concat two arrays instead of pushing all elements.

Screenshot 2023-04-21 at 14 32 12

### PR DESCRIPTION
**Pull request by vlats05**
close #983

### CODE REVIEW

**Comment by yoyo837** [comment]
Why it can resolve `Maximum call stack size exceeded error`?

**Comment by vlats05** [comment]
@yoyo837 
actually max number of elements in array.push depends on many things, including javascript engine and browser version, optimizations etc.
Also there are many topics describing the same issue:
https://github.com/nodejs/node/issues/27732
https://stackoverflow.com/questions/42263108/why-does-apply-with-too-many-arguments-throw-maximum-call-stack-size-exceeded

Made a simple example on codesandbox:
https://codesandbox.io/s/determined-gates-dkku0x

File: src/hooks/useFlattenRecords.ts [hunk 1]
```diff
@@ -60,13 +60,15 @@ export default function useFlattenRecords<T>(
...
     if (expandedKeys?.size) {
-      const temp: { record: T; indent: number; index: number }[] = [];
+      let temp: { record: T; indent: number; index: number }[] = [];
 
       // collect flattened record
       for (let i = 0; i < data?.length; i += 1) {
         const record = data[i];
 
-        temp.push(...flatRecord<T>(record, 0, childrenColumnName, expandedKeys, getRowKey, i));
+        temp = temp.concat(
```
Comment (yoyo837) [inline src/hooks/useFlattenRecords.ts:69]: Please add some comments to keep holding this change.

**Comment by vlats05** [comment]
@yoyo837 Can you merge it? I don't have write permissions.

**Pull request merged**